File: backend/src/main/java/com/rental/car/Models/Disponibility.java
package com.rental.car.Models;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.Data;

import java.sql.Timestamp;

@Entity
@Data
public class Disponibility {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "car")
    private Long carId;
    @Column(name = "user")
    private Long userId;
    @Column(name = "datein")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Timestamp dateIn;
    @Column(name = "dateout")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Timestamp dateOut;
    private Boolean paid;
}

File: frontend/src/models/Disponibility.ts
export class Disponibility{
  public id:number|undefined;
  public userId:number|undefined;
  public carId:number;
  public dateIn:string;
  public dateOut:string;
  public paid?:boolean;

  constructor(carId:number,dateIn:string,dateout:string,paid?:boolean,userId?:number){
    this.userId = userId;
    this.carId = carId;
    this.dateIn = dateIn;
    this.paid = paid;
    this.dateOut = dateout;
  }
}